app/controllers/user.js

```diff
@@ -16,12 +16,38 @@ const passwordValid = password => {
   return isAlphanumeric.test(password) && password.length >= 8;
 };
 
+const generateUser = (user, next) => {
+  if (!emailValid(user.email)) {
+    logger.error(`Email: ${user.email} invalid.`);
+    return next(errors.emailNotValid(user.email));
```

Comment:
this function will sometimes return a promise and other cases nothing. So you will have some unexpected errores in the background. 
It's better if you throw the error instead of calling next. So this function will always return a promise, sometimes fulfilled and sometimes rejected. Also you leave the calling function to handle that error with the next middleware call.
So in this line you should leave `throw errors.emailNotValid(user.email)` or `return Promise.reject(errors.emailNotValid(user.email))` and remove the next param of this function.